src/topology/metric_space/basic.lean

```diff
@@ -1304,11 +1304,23 @@ def uniform_inducing.comap_pseudo_metric_space {α β} [uniform_space α] [pseud
   (f : α → β) (h : uniform_inducing f) : pseudo_metric_space α :=
 (pseudo_metric_space.induced f ‹_›).replace_uniformity h.comap_uniformity.symm
 
-instance subtype.psudo_metric_space {α : Type*} {p : α → Prop} [t : pseudo_metric_space α] :
-  pseudo_metric_space (subtype p) :=
-pseudo_metric_space.induced coe t
+instance subtype.pseudo_metric_space {p : α → Prop} : pseudo_metric_space (subtype p) :=
+pseudo_metric_space.induced coe ‹_›
 
-theorem subtype.pseudo_dist_eq {p : α → Prop} (x y : subtype p) : dist x y = dist (x : α) y := rfl
+theorem subtype.dist_eq {p : α → Prop} (x y : subtype p) : dist x y = dist (x : α) y := rfl
+theorem subtype.nndist_eq {p : α → Prop} (x y : subtype p) : nndist x y = nndist (x : α) y := rfl
+
+namespace mul_opposite
+
+@[to_additive]
+instance : pseudo_metric_space (αᵐᵒᵖ) := pseudo_metric_space.induced mul_opposite.unop ‹_›
```

Comment:
Do we have `topological_space` and/or `uniform_space` instances on `mul_opposite`? If not, then we should. If yes, then please verify that there is no non-defeq diamond here.

Reply:
We had `topological_space`, but not `uniform_space`. I added the latter.

> If yes, then please verify that there is no non-defeq diamond here.

I'd rather skip this step - the approach I'm using is the same as the `subtype` instances, and I'm going to assume whoever added those either already did the test, or decided it didn't matter.